src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs

```diff
@@ -197,31 +197,29 @@ private void AddDeleteOperation(OpenApiPathItem item, NavigationPropertyRestrict
             Debug.Assert(!LastSegmentIsRefSegment);
 
             DeleteRestrictionsType navPropDeleteRestriction = restriction?.DeleteRestrictions ??
-               Context.Model.GetRecord<DeleteRestrictionsType>(NavigationProperty);
-
+               Context.Model.GetRecord<DeleteRestrictionsType>(NavigationProperty);
+
             if (NavigationProperty.ContainsTarget)
             {
-                DeleteRestrictionsType entityDeleteRestriction = Context.Model.GetRecord<DeleteRestrictionsType>(_navPropEntityType);
-                bool isDeletableDefault = navPropDeleteRestriction == null && entityDeleteRestriction == null;
-
-                if (isDeletableDefault ||
-                ((entityDeleteRestriction?.IsDeletable ?? true) &&
-                (navPropDeleteRestriction?.IsDeletable ?? true)))
-                {
-                    if (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many || LastSegmentIsKeySegment)
-                    {
-                        AddOperation(item, OperationType.Delete);
-                    }
-                }
+                DeleteRestrictionsType entityDeleteRestriction = Context.Model.GetRecord<DeleteRestrictionsType>(_navPropEntityType);
+                bool isDeletableDefault = navPropDeleteRestriction == null && entityDeleteRestriction == null;
+
+                if ((isDeletableDefault ||
+                ((entityDeleteRestriction?.IsDeletable ?? true) &&
+                (navPropDeleteRestriction?.IsDeletable ?? true))) &&
+                (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many ||
+                LastSegmentIsKeySegment))
+                {
+                    AddOperation(item, OperationType.Delete);
+                }
             }
-            else
-            {
-                if ((navPropDeleteRestriction?.IsDeletable ?? false) &&
-                    (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many ||
-                    LastSegmentIsKeySegment))
-                {
-                    AddOperation(item, OperationType.Delete);
-                }
+            else if ((navPropDeleteRestriction?.Deletable ?? false) &&
+                    (NavigationProperty.TargetMultiplicity() != EdmMultiplicity.Many ||
+                    LastSegmentIsKeySegment))

```

Comment:
this seems to be a requirement for both blocks, what about extracting that statement, negating it and returning early before those two blocks to improve readability?

Reply:
Resolved